### DISCUSSION TITLE
Antichess endgame really struggles in certain instances

### DISCUSSION THREAD

**Discussion by awkwardmammal2** [fairy-stockfish/Fairy-Stockfish#729] [Ideas]
Tbh idk how to fix it, i thought I'd just bring it up for discussion, fairy stockfish seems to really struggle with endgames, particularly when it has a few pawns left, it takes a longgggg time to consider pushing them forward to promote, which is usually the only way to win.

**Comment by ianfab** [comment]
Could you please provide FENs to reproduce this behavior? Otherwise it is hard to follow what we are talking about.

**Reply by awkwardmammal2** [reply]
instances like this: 5R2/3P2N1/8/4K1P1/8/8/2k4P/8 b - - 24 61
It seems to just want to bounce around the non-pawn pieces when clearly the only path to winning would be to get the pawns promoted.
The issue starts at earlier moves https://lichess.org/study/twn65LLp it's in my e3 e6 chapter

**Reply by ianfab** [reply]
Thanks. To my understanding the issue just is that it unnecessarily delays the win, but doesn't spoil it at any point. If that is what you mean from a developer perspective that is kind of a non-issue.

I understand that such behavior can be annoying or confusing, but engines are optimized for playing strength, i.e., game results, not style, so unless such an observation leads to an idea for a patch that directly improves playing strength there isn't much that can be done about it.

**Reply by awkwardmammal2** [reply]
As far as I can tell it is simply unable to find the win, not just delays it, so if the 50 move rule is applied then it does result in a draw when a position like that should be a win. 8/2NP2PP/6R1/5K2/8/8/3k4/8 w - - 49 110 for me it shows +8.5, but if you manually create a rook with the h pawn then it'll basically go, oh right, I can promote, and can find finally find wins. I've only had success in manually going in to promote a piece in many cases like that. Unfortunately I do not have any knowledge of the code used, my coding background is only based in engineering, so def not geared towards something like this, but would be interested in maybe learning so I'd be able to actually suggest patches for this. Though something I could suggest even now is shaving an option to toggle having a 50 move limit (or maybe multiple numerical options or option for the user to input a number) for antichess. Though would that be of any use in many cases given the high number of permutations?

**Reply by ianfab** [reply]
The 50 move rule always has been in place and working for antichess as well. However, in the FEN it is half-moves, so e.g. 99, not 49, would indicate that the very next move triggers it. If you test it on `8/2NP2PP/6R1/5K2/8/8/3k4/8 w - - 99 110` it immediately goes for a promotion. So I assume that as expected it is just a usability concern, and nothing that impacts actual game play.

**Reply by awkwardmammal2** [reply]
alr, sounds good then
